crates/modelardb_server/src/storage/uncompressed_data_manager.rs

```diff
@@ -608,7 +787,7 @@ impl UncompressedDataManager {
         .unwrap();
 
         // unwrap() is safe as lock() only returns an error if the lock is poisoned.
-        let freed_memory = data_buffer.memory_size();
+        let freed_memory = UncompressedInMemoryDataBuffer::memory_size();
```

Comment:
Fixed by returning `memory_use` and `disk_use` from the pattern matching and returning `memory_use` to the memory pool and writing `memory_use` and `disk_use` to the relevant metrics. To ensure over reservation of memory never happens the compressor should also reserve memory when reading an on-disk buffer into memory to compress it. However, it is unclear if this can lead to deadlocks due to the uncompressed manager waiting for buffers to be compressed and the compressor waits for uncompressed memory to be returned to the pool. So since this PR is already very large, handling these edge cases are part of the task to refine memory management in the storage engine.